lib/fluent/plugin/buf_file_single.rb

```diff
@@ -241,6 +241,20 @@ def handle_broken_files(path, mode, e)
         File.unlink(path) rescue nil
       end
 
+      def evacuate_chunk(chunk)
+        unless chunk.is_a?(Fluent::Plugin::Buffer::FileSingleChunk)
```

Comment:
If you need to check strictly, you should use `instance_of?`.

```
irb(main):001> class A; end
=> nil
irb(main):002> class B < A; end
=> nil
irb(main):003> obj = B.new
=> #<B:0x00007ce071d3cc18>
irb(main):004> obj.is_a?(B)
=> true
irb(main):005> obj.is_a?(A)
=> true
irb(main):006> obj.instance_of?(B)
=> true
irb(main):007> obj.instance_of?(A)
=> false
```

`is_a?` method passes the inherited class as well.